src/main/java/appeng/tile/misc/TileCondenser.java

```diff
@@ -223,65 +241,174 @@ public boolean canExtractItem( final int slotIndex, final ItemStack extractedIte
 	}
 
 	@Override
-	public int fill( final EnumFacing from, final FluidStack resource, final boolean doFill )
+	public void updateSetting( final IConfigManager manager, final Enum settingName, final Enum newValue )
 	{
-		if( doFill )
-		{
-			this.addPower( ( resource == null ? 0.0 : (double) resource.amount ) / 500.0 );
-		}
-
-		return resource == null ? 0 : resource.amount;
+		this.addPower( 0 );
 	}
 
 	@Override
-	public FluidStack drain( final EnumFacing from, final FluidStack resource, final boolean doDrain )
+	public IConfigManager getConfigManager()
 	{
-		return null;
+		return this.cm;
 	}
 
-	@Override
-	public FluidStack drain( final EnumFacing from, final int maxDrain, final boolean doDrain )
+	public double getStoredPower()
 	{
-		return null;
+		return this.storedPower;
 	}
 
-	@Override
-	public boolean canFill( final EnumFacing from, final Fluid fluid )
+	private void setStoredPower( final double storedPower )
 	{
-		return true;
+		this.storedPower = storedPower;
 	}
 
 	@Override
-	public boolean canDrain( final EnumFacing from, final Fluid fluid )
+	public boolean hasCapability( Capability<?> capability, EnumFacing facing )
 	{
-		return false;
+		if( capability == CapabilityItemHandler.ITEM_HANDLER_CAPABILITY )
+		{
+			return true;
+		}
+		else if( capability == CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY )
+		{
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This branch's code block is the same as the block for the branch on line 269. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1871)